### ISSUE TITLE
Clinical ResultCell display does not update between value and view result message on resize

### ISSUE THREAD

**Issue by SharathAlur** [cerner/terra-clinical#762]
# Bug Report

## Description

Clinical ResultCell display does not update between value and view result message on resize

## Steps to Reproduce

1. Enter a large value to display in cell
2. Resize the screen
3.
4.

## Additional Context / Screenshots

https://user-images.githubusercontent.com/22268936/122743921-1fce9080-d2a5-11eb-8109-df9f2c54cb1c.mov

## Expected Behavior

The result cell should display value or view result message based on the cell size on resize.

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

**Comment by jmsv6d** [comment]
@SharathAlur after debugging your project locally we discovered that there is at least on location where the `isNumeric` prop is not being applied appropriately to the `FlowsheetResultCell`. The `isNumeric` prop must be set in order to trigger the "View Results" behavior. This is the location that it was discovered to be missing the prop:  https://github.cerner.com/Assessments/clinical-flowsheet-js/blob/master/src/clinical-flow-sheet-js/store/selectors/selectors.ts#L372-L406
You will want to verify that there are no other locations where this prop is missing.
Another thing we found was that the `FlowsheetResultCell` does the calculation to determine if the "View Results" message should display on render. Your project resizes the grid columns as the page resizes. The change of width on the DOM does not trigger the `FlowsheetResultCell` to re-render. It would not be performant for the cells to each contain a resize observer to detect that it's size has changed, it would be best to determine that a resize has occurred higher up. The `FlowsheetResultCell` is not coupled to the `DataGrid` so there is not a great way for them to communicate a change of column width. In the future we believe that we should have the cells be an internal component to the grid that it should be used in, but for now we recommend that you force the `FlowsheetResultCell`s with numeric values to update when the columns are resized. You could do this by updating the components' `key`.

**Comment by SharathAlur** [comment]
Updating the view result message on resize issue is still present even after setting isNumeric to true.
If I have the screen size smaller one initially, where result could not be shown and it shows 'view result' instead. But even after resizing the screen to display the data, the 'view result' message still appears. It does not re-calculates the available space and display the value.

https://user-images.githubusercontent.com/22268936/127274218-87319070-c59f-4f46-a9fa-183883e5a603.mov

**Comment by jmsv6d** [comment]
Did you also make the other suggested change?

> Another thing we found was that the FlowsheetResultCell does the calculation to determine if the "View Results" message should display on render. Your project resizes the grid columns as the page resizes. The change of width on the DOM does not trigger the FlowsheetResultCell to re-render. It would not be performant for the cells to each contain a resize observer to detect that it's size has changed, it would be best to determine that a resize has occurred higher up. The FlowsheetResultCell is not coupled to the DataGrid so there is not a great way for them to communicate a change of column width. In the future we believe that we should have the cells be an internal component to the grid that it should be used in, but for now we recommend that you force the FlowsheetResultCells with numeric values to update when the columns are resized. You could do this by updating the components' key.

**Comment by SharathAlur** [comment]
My Bad, I missed to go through this part. Yeah If we re-render the data it is working fine. Thanks @jmsv6d .
This issue can be closed.

**Comment by ryanthemanuel** [comment]
Closing per above comments.
